src/CookieCollection.php

```diff
@@ -0,0 +1,298 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Yiisoft\Yii\Web;
+
+use ArrayAccess;
+use ArrayIterator;
+use Closure;
+use Countable;
+use DateTimeImmutable;
+use InvalidArgumentException;
+use IteratorAggregate;
+use Psr\Http\Message\ResponseInterface;
+use Psr\Http\Message\ServerRequestInterface;
+
+use function array_keys;
+use function array_values;
+use function count;
+use function in_array;
+
+/**
+ * A CookieCollection is a collection implementation to wrap an array of Cookie class instances
+ */
+final class CookieCollection implements IteratorAggregate, ArrayAccess, Countable
```

Comment:
The use case overall isn't clear.

Reply:
Yep, I'm working rewriting some of the php doc blocks with more clear texts. And also adding php docs where they are absent in both classes. I'm not a native English speaker, and don't have sufficient experience writing documentation, so pardon any mistakes there.

Reply:
Me neither. Eventually natives may correct us. That's fine.